lib/libzpool/zrepl_mgmt.c

```diff
@@ -412,3 +415,164 @@ uzfs_zvol_store_last_committed_io_no(zvol_state_t *zv, char *key,
 	VERIFY0(uzfs_update_zap_entries(zv,
 	    (const uzfs_zap_kv_t **) kv_array, 1));
 }
+
+void
+uzfs_dump_zvol_stats(nvlist_t **stat)
+{
+	zvol_info_t *zinfo = NULL;
+	nvlist_t *zstats, *nv, **nrebuild_stats;
+	int i, rebuild_count = 0;
+	rebuild_stats_t *r_stats;
+	struct sockaddr_in sock_addr;
+	socklen_t addrlen = sizeof (struct sockaddr);
+	char tbuf[50];
+	uzfs_mgmt_conn_t *conn;
+
+	ASSERT(!*stat);
+
+	if (nvlist_alloc(&zstats, NV_UNIQUE_NAME, 0)) {
+		LOG_ERR("failed to alloc nvlist\n");
+		return;
+	}
+
+	(void) mutex_enter(&zvol_list_mutex);
```

Comment:
Not related to your changes, but this lock seems to be little heavy now as new use cases emerging ...

Reply:
yes.